.gitlab-ci.yml

```diff
@@ -19,26 +18,42 @@ variables:
 cache:
   paths:
     - .cache/pip
-    - venv/
 
-before_script:
-  - python -V  # Print out python version for debugging
-  - pip install -r requirements.txt
-  - pip install python-coveralls
-
-test:
+.tests:
   stage: test
   script:
-    - echo $TEST_SUITE
-    - echo $TEST_SUITE > test.sh
-    - bash test.sh
+    - python -V  # Print out python version for debugging
+    - pip install -r requirements.txt
+    - pip install python-coveralls
+    - python -m coverage run -m pytest
     - coverage report
     # Test that it can be started using the checked-out source code:
     - python -m pyega3.pyega3 --help
     # Test that it can be started using the installed package:
     - pip install -e . && pyega3 --help
   #  - coveralls
 
+test_python36:
+  extends: .tests
+  image: python:3.6
+
+test_python37:
+  extends: .tests
+  image: python:3.7
+
+test_python38:
+  extends: .tests
+  image: python:3.8
+
+test_python39:
+  extends: .tests
+  image: python:3.9
+
+test_python310:
+  extends: .tests
+  image: python:3.10
+
```

Comment:
This way the tests are running under different versions of Python. Using this technique, two bugs were caught:
* pytest 6.0.1 doesn't run under Python 3.10
* One of the tests didn't run under Python 3.6 and 3.7

Reply:
Originally, I only wanted to update the README. But when the automated tests ran in GitLab, they failed, because our (original) GitLab tests used the latest Python version (currently 3.10) and one of our old dependencies (pytest 6.0.1) didn't support Python 3.10. So I had to upgrade that old dependency. And then I created these tests to avoid such a bug in the future. And then these tests caught another bug (=one of the tests didn't run under Python 3.6 and 3.7), so I had to modify that test. That's why this commit blew up this big.